core/src/main/java/com/linecorp/armeria/internal/common/AbstractKeepAliveHandler.java

```diff
@@ -125,9 +134,31 @@ public final void initialize(ChannelHandlerContext ctx) {
         }
         isInitialized = true;
 
+        final ConnectionEventKey connectionEventKey = connectionEventKey(channel);
+
+        try {
+            connectionEventListener.connectionOpened(connectionEventKey, channel);
```

Comment:
If `NoopKeepAliveHandler` is used, this event isn't called anymore. https://github.com/line/armeria/blob/89e0257050e5d6099675c853b48b269ec19aa9e7/core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java#L98
Is there any reason that you've moved this logic from `HttpChannelPool`?

Reply:
I tried to move open and close event trigger logic into the AbstractKeepAliveHandler to reuse this on the server side. However, as you said, I had to make extra changes to do that. Any feedback for this? @minwoox @ikhoon 

- Adding connection opened and closed event trigger logic into the NoopKeepAliveHandler
- Handle HTTP Upgrade cases with context-checking(HttpProtocolUpgradeHandler)

Reply:
> I tried to move open and close event trigger logic into the AbstractKeepAliveHandler to reuse this on the server side

Since the connection mechanisms differ between the server-side and client-side, I don't think we have to put the logic into the same class.
Instead, we could consider incorporating the logic for the server-side into ConnectionLimitingHandler, which is already used for managing connections. You can find the ConnectionLimitingHandler implementation here: [Link](https://github.com/line/armeria/blob/9061e241961c806951dd855be65fd871131a1835/core/src/main/java/com/linecorp/armeria/server/ConnectionLimitingHandler.java#L60)